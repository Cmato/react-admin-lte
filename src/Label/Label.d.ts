import * as React from 'react';
import { PropTypes } from '..';

export interface LabelProps {
  backgroundColor: PropTypes.BackgroundColor;
}

declare const Label: React.ComponentType<LabelProps>;

export default Label;
