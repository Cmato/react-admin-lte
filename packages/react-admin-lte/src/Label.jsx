import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import { Label as BsLabel } from 'react-bootstrap';

const propTypes = {
  background: PropTypes.oneOf(['light-blue', 'red', 'green', 'yellow']),
  className: PropTypes.string,
  children: PropTypes.node,
};

const Label = ({
  background,
  className,
  children,
}) => {
  const classes = {
    label: true,
  };

  if (background) {
    classes[`bg-${background}`] = true;
  }

  return (
    <BsLabel bsClass={classNames(className, classes)}>{children}</BsLabel>
  );
};

Label.propTypes = propTypes;

export default Label;
