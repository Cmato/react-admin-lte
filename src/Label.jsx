import React from 'react';
import classNames from 'classnames';
import { Label as BsLabel } from 'react-bootstrap';

const propTypes = {
  background: React.PropTypes.oneOf(['light-blue', 'red', 'green', 'yellow']),
  className: React.PropTypes.string,
  children: React.PropTypes.node,
};

const Label = ({
  background,
  className,
  children,
}) => {
  const classes = {
    label: true,
  };

  if (background) {
    classes[`bg-${background}`] = true;
  }

  return (
    <BsLabel bsClass={classNames(className, classes)}>{children}</BsLabel>
  );
};

Label.propTypes = propTypes;

export default Label;
