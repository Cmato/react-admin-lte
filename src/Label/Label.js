import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import colorProp from '../utils/colorProp';

/**
 * `Label` allows for specifying a background color. To use a contextually
 * styled label use the react-bootstrap component.
 */
const Label = ({
  backgroundColor,
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'label': true,
    [`bg-${backgroundColor}`]: true
  }, classNameProp);

  return <span className={className} {...other}>{children}</span>;
};

Label.propTypes = {
  /**
   * Color of background.
   */
  backgroundColor: colorProp.isRequired,
  /**
   * The content of the component.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
};

export default Label;
