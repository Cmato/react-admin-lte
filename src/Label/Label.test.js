import React from 'react';
import { shallow } from 'enzyme';
import Label from './Label';

it('renders correctly', () => {
  expect(shallow((
    <Label
      backgroundColor="red"
    >
      <div />
    </Label>
  ))).toMatchSnapshot();
});

