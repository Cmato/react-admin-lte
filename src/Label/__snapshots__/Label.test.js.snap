// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="label bg-red"
>
  <div />
</span>
`;
